src/plugins/graph/models/graph-model.ts

```diff
@@ -765,21 +764,48 @@ export const GraphModel = TileContentModel
         }
       ));
     },
-    setDataConfigurationReferences() {
-      // Updates pre-existing DataConfiguration objects that don't have the now-required references
-      // for dataset and metadata. We can determine these from the unique shared models these
-      // legacy tile models should have.
+    initializeSharedModelReferences() {
       const smm = getSharedModelManager(self);
       if (smm && smm.isReady) {
         const sharedDataSets = smm.getTileSharedModelsByType(self, SharedDataSet);
+        let sharedMetadata = smm.getTileSharedModelsByType(self, SharedCaseMetadata);
+
+        // If there's a shared dataset without corresponding shared case metadata, create a new shared case
+        // metadata instance, link it to the dataset, and add it to the tile. This is needed when graph tiles are
+        // copied since the original graph tile's case metadata is not copied along with the shared dataset.
+        sharedDataSets.forEach((sds) => {
+          if (!isSharedDataSet(sds)) return;
+          const hasLinkedCaseMetadata = sharedMetadata.some((smd) => {
+            if (isSharedCaseMetadata(smd)) {
+              return smd.data === sds.dataSet;
+            }
+          });
+          if (!hasLinkedCaseMetadata) {
+            const smd = SharedCaseMetadata.create();
+            smd.setData(sds.dataSet);
+            smm.addTileSharedModel(self, smd);
+            const datasetLayer = self.layers.find((layer) => layer.config.dataset === sds.dataSet);
+            if (datasetLayer) {
+              datasetLayer.config.metadata = smd;
+            }
```

Comment:
@bgoldowsky @scytacki The `legacyGraph` check on lines 797-799 broke the fix we had working. I found adding these lines (787-790) made it work again. Apparently, line 812 was taking care of that before we added the `legacyGraph` check.

Is it correct to add the shared case metadata instance like that to the same layer as the shared dataset?

Reply:
Yes, that looks correct to me.